package/src/animation/functions/interpolate.ts

```diff
@@ -1,169 +1,182 @@
-// eslint-disable-next-line max-len
-// Taken from https://github.com/facebook/react-native/blob/0b9ea60b4fee8cacc36e7160e31b91fc114dbc0d/Libraries/Animated/src/nodes/AnimatedInterpolation.js
-
-type ExtrapolateType = "extend" | "identity" | "clamp";
-
-function interpolateFunction(
-  input: number,
-  inputRange: [number, number],
-  outputRange: [number, number],
-  options: {
-    easing: (input: number) => number;
-    extrapolateLeft: ExtrapolateType;
-    extrapolateRight: ExtrapolateType;
-  }
-): number {
-  const { extrapolateLeft, extrapolateRight, easing } = options;
-
-  let result = input;
-  const [inputMin, inputMax] = inputRange;
-  const [outputMin, outputMax] = outputRange;
-
-  if (result < inputMin) {
-    if (extrapolateLeft === "identity") {
-      return result;
-    }
-
-    if (extrapolateLeft === "clamp") {
-      result = inputMin;
-    } else if (extrapolateLeft === "extend") {
-      // noop
-    }
-  }
-
-  if (result > inputMax) {
-    if (extrapolateRight === "identity") {
-      return result;
-    }
-
-    if (extrapolateRight === "clamp") {
-      result = inputMax;
-    } else if (extrapolateRight === "extend") {
-      // noop
-    }
-  }
-
-  if (outputMin === outputMax) {
-    return outputMin;
-  }
-
-  // Input Range
-  result = (result - inputMin) / (inputMax - inputMin);
+/* eslint-disable max-len */
+export enum Extrapolate {
```

Comment:
To avoid having to import `Extrapolate` when using it, should the type just for simplicity be a typed string value like:

```ts
type Extrapolate = "IDENTITY" | "CLAMP" | "EXTEND";
```

Then the type doesn't need to be exported and we still get intellisense when using the `interpolate` function:

```ts
interpolate(input, output, "CLAMP");
```